test/nuxeo-edit-documents-button.test.js

```diff
@@ -0,0 +1,434 @@
+    await flush();
+  }
+};
+
+// XXX Copied from https://github.com/nuxeo/nuxeo-elements/blob/maintenance-3.0.x/ui/test/nuxeo-document-picker.test.js
+// should be refactored and moved to the public helpers in nuxeo-elements
+const waitForDialogClose = async (dialog) => {
+  if (isElementVisible(dialog)) {
+    await waitForEvent(dialog, 'iron-overlay-closed');
+    await flush();
+  }
+};
+
+// determine base module path (relies on @open-wc/webpack-import-meta-loader)
+const { url } = import.meta;
+const base = url.substring(0, url.lastIndexOf('/'));
+const baseUrl = `${base}/layouts/`;
+
+suite('nuxeo-edit-documents-button', () => {
+  let button;
+
+  const buildButton = async (layoutId = 'default') => {
+    const documents = ['the content of this array is irrelevant'];
+    const actionButton = await fixture(
+      html`
+        <nuxeo-edit-documents-button href-base="${baseUrl}" layout="${layoutId}" .documents="${documents}">
+        </nuxeo-edit-documents-button>
+      `,
+    );
+    const bulkLayout = actionButton.$$('nuxeo-layout');
+    if (!bulkLayout.element) {
+      await waitForLayoutLoad(bulkLayout);
+    }
+    await flush();
+    return actionButton;
+  };
+
+  const getWidget = (xpath) => button._getBoundElementFromPath(`document.properties.${xpath}`);
+  const getBulkWidget = (boundElement) => button._getBulkWidget(boundElement);
+
+  suite('Default Layout', () => {
+    test('Should open, close and reopen the bulk edit action dialog', async () => {
+      button = await buildButton();
+      const { dialog } = button.$;
+      const cancelButton = button.$$('paper-button.secondary');
+      // check that dialog is closed
+      expect(dialog.opened).to.be.false;
+      expect(isElementVisible(dialog)).to.be.false;
+      expect(isElementVisible(cancelButton)).to.be.false;
+      // open the bulk edit dialog
+      button._execute();
+      await waitForDialogOpen(dialog);
+      // check that dialog is open
+      expect(dialog.opened).to.be.true;
+      expect(isElementVisible(dialog)).to.be.true;
+      expect(isElementVisible(cancelButton)).to.be.true;
+      // click the cancel button
+      cancelButton.click();
+      await waitForDialogClose(dialog);
+      // check that dialog is closed
+      expect(dialog.opened).to.be.false;
+      expect(isElementVisible(dialog)).to.be.false;
+      expect(isElementVisible(cancelButton)).to.be.false;
+      // reopen the bulk edit dialog
+      button._execute();
+      await waitForDialogOpen(dialog);
+      // check that dialog is open
+      expect(dialog.opened).to.be.true;
+      expect(isElementVisible(dialog)).to.be.true;
+      expect(isElementVisible(cancelButton)).to.be.true;
+    });
+
+    test('Should disable the save button if there are no properties to update', async () => {
+      button = await buildButton();
+      const { save } = button.$;
+      const expiredWidget = getWidget('dc:expired');
+      // open the bulk edit dialog
+      button._execute();
```

Comment:
We should avoid relying on private methods as much as possible, given that they expose internal implementation details. I would expect us to be able to achieve the same thing with:
```suggestion
      button.click();
```

Reply:
Unfortunately `button.click()` doesn't map to `button._execute()`, but `button.$$('.action').click()` does.
I'm going to use that instead!